compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -290,15 +282,28 @@ FailureOr<ParameterSetting> ParameterSetting::create(
       /*vectorK=*/k1Pack};
   TileSize maxL1Size = selectL1TileSizes(tileParams);
 
+  // For pack-peel pipeline, the first level tile size has to be numRows/numCols
+  // times of the L1 tile size.
   uint32_t maxL0SizeM = numRows * maxL1Size.M;
   uint32_t maxL0SizeN = numCols * maxL1Size.N;
   uint32_t M0 = findLargestFactor(M, maxL0SizeM, maxL1Size.M);
   uint32_t N0 = findLargestFactor(N, maxL0SizeN, maxL1Size.N);
   uint32_t m0Pack = (M0 / numRows) % m1Pack == 0 ? (M0 / numRows) : M0;
   uint32_t n0Pack = (N0 / numCols) % n1Pack == 0 ? (N0 / numCols) : N0;
 
-  // In pack-peel pipeline there is only one level of tiling for K dimension,
-  // so set K1 = 0. The packed outer K dimension needs to be 1, so set K0 = 1.
+  // For pack-peel-4-level-tiling pipeline, we use the largest tile sizes that
+  // can fit in the MemTile memory.
+  if (kPackScaleL1 == 2 && isObjectFifo) {
+    tileParams.memoryLimit = deviceModel.getMemTileSizeInBytes() * numCols;
+    tileParams.bufferDepthAcc =
+        isMatmulWithElementwiseConsumer(linalgOp) ? 0 : 2;
```

Comment:
Why is `bufferDepthAcc` equal to `0` if we have an elementwise consumer? We should still need an accumulator in that case so should be `1`?

Reply:
This is just a trick to remove the memory computation for matmul output since the real output is from elementwise op in this case.

Reply:
> output buffer from L1 to L2 is only from elementwise op
Even though that's correct, there is still an internal accumulator buffer used (depth 1), so I don't think this is correct and you should still take into account that buffer as well.

Reply:
I understand there's an accumulator buffer in L1, but why there's also one in L2? There's some preserved memory space for that?

Reply:
Okay, I think it's easy to get confused because I was reusing the same `TileParams` struct when calculating L2 memory usage. 

`tileParams.bufferDepthAcc` should not always be 0, because we have to use it for pure matmul. In the pure matmul case, because `nBytesElemOut` = 0, it makes this to be zero regardless of `bufferDepthC` value.
`uint32_t C = params.numBytesC * m * n * params.bufferDepthC;`

While for `matmul + elementwise` case, I have to use the trick to make this zero:
`uint32_t Acc = params.numBytesAcc * m * n * params.bufferDepthAcc;
`

Reply:
> tileParams.bufferDepthAcc should not always be 0, because we have to use it for pure matmul. In the pure matmul case, because nBytesElemOut = 0, it makes uint32_t C = params.numBytesC * m * n * params.bufferDepthC to be zero regardless of bufferDepthC value. So we have to keep bufferDepthAcc for the output buffer.

The use of `Acc` for L2 is confusing, I think it would be more clear to always use `nBytesElemOut` / `bufferDepthC ` for both L1 and L2 and set `bufferDepthAcc` to `0` for L1 if not followed by an elementwise operation.

Reply:
I've made changes as suggested. Also removed `Acc` in L2 memory calculation.